c++/src/StripeStream.cc

```diff
@@ -37,7 +38,8 @@ namespace orc {
         stripeStart_(stripeStart),
         input_(input),
         writerTimezone_(writerTimezone),
-        readerTimezone_(readerTimezone) {
+        readerTimezone_(readerTimezone),
+        readCache_(reader.getReadCache()) {
```

Comment:
You might directly call `RowReaderImpl.getFileContents()` to get readCache as suggested above.